File: 프로그래머스/0/181894. 2의 영역/2의 영역.py
def solution(arr):
    answer = []
    if 2 in arr:
        idx = []
        for i in range(len(arr)):
            if arr[i] == 2:
                idx.append(i)
        if len(idx) == 1:
            answer = [2]
        else:
            answer = arr[idx[0]: idx[-1]+1]
    else:
        answer.append(-1)
    return answer

File: 프로그래머스/0/181894. 2의 영역/2의 영역.java
import java.util.ArrayList;

class Solution {
    public int[] solution(int[] arr) {
        ArrayList<Integer> idx = new ArrayList();
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == 2) {
                idx.add(i);
            }
        }
        if (idx.size() != 0) {
            int start = idx.get(0), end = idx.get(idx.size() -1);
            int[] answer = new int[end - start + 1];
            for (int i = start; i<= end; i++) {
                answer[i - start] = arr[i];
            }
            return answer;
        }
        int[] answer = new int[1];
        answer[0] = -1;
        return answer;
    }
}